src/datasets/keyhash.py

```diff
@@ -70,10 +70,10 @@ def __init__(self, hash_data):
 class DuplicatedKeysError(Exception):
     """Raise an error when duplicate key found."""
 
-    def __init__(self, key):
-        self.prefix = "FAILURE TO GENERATE DATASET !"
-        self.err_msg = f"\nFound duplicate Key: {key}"
-        self.suffix = "\nKeys should be unique and deterministic in nature"
+    def __init__(self, key, duplicate_key_indices):
+        self.prefix = f"Found multiple examples generated with the same key"
+        self.err_msg = f"\nThe following examples {', '.join(duplicate_key_indices)} have the key {key}"
+        self.suffix = "\nPlease fix the dataset script at <Path to Dataset> to avoid duplicate keys"
```

Comment:
I think the suffix can simply be appended in `builder.py` instead of using `.replace()`

Reply:
Changed accordingly.